options.go

```diff
@@ -121,6 +123,10 @@ type Options struct {
 	// NoNetworkFacts disables built-in network interface facts gathering
 	NoNetworkFacts bool `json:"no_network_facts,omitempty"`
 
+	// os.Args opt-out https://github.com/choria-io/machine-room/issues/8
+	CustomArgsEnabled bool     `json:"-"`
+	CustomArgs        []string `json:"-"`
```

Comment:
@ripienaar Thanks for merging this, but I was wondering: Is the boolean unnecessary? Could I have just checked if CustomArgs is `nil`?

At least this is a bit more readable, so won't change it unless you want me to.

Reply:
Yeah you're right, checking `len(CustomArgs)==0` would be better

Reply:
Cool let me fix it

Reply:
I pushed 2 dependency updates - one also adding a new watcher

https://choria.io/docs/autoagents/watcher_reference/#expression-watcher

Reply:
Gonna check nil, len and cap are 0 for a nil slice.

```go
package main

import "fmt"

func main() {
	var args []string
	p("args", args)
	fmt.Println()
	p("args2", []string{})
}

func p(name string, args []string) {
	if args == nil {
		fmt.Println(name, "is nil")
	} else {
		fmt.Println(name, "is NOT nil")
	}
	fmt.Printf("%s: %+v\n", name, args)
	fmt.Printf("len(%s): %+v\n", name, len(args))
	fmt.Printf("cap(%s): %+v\n", name, cap(args))
}
```

```
❯ go run foo.go
args is nil
args: []
len(args): 0
cap(args): 0

args2 is NOT nil
args2: []
len(args2): 0
cap(args2): 0
```

Reply:
len()==0 is enough. Nil slice is 0 len.

Reply:
I left it as a nil check because I think we want to disambiguate between the case where the user passes in args of length 0 and the case where the user didn't pass in any args at all.  In the second case we want to use os.Args.

In other words, len() will always be 0 when the user isn't me...  See what I'm getting at?